helpers/bsc.sh

```diff
@@ -0,0 +1,152 @@
+# Copyright (C) 2019 Evgeny Golyshev <eugulixes@gmail.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Checks if it's possible to connect to the Redis server, specified via the
+# Pieman parameters.
+# Globals:
+#     ENABLE_BSC_CHANNEL
+#     PIEMAN_UTILS_DIR
+#     PYTHON
+#     REDIS_HOST
+#     REDIS_IS_AVAILABLE
+#     REDIS_PORT
+# Arguments:
+#     None
+# Returns:
+#     None
+check_redis() {
+    if ${ENABLE_BSC_CHANNEL}; then
+        info "checking if it is possible to connect to the Redis server" \
+             "${REDIS_HOST}:${REDIS_PORT}"
+
+        if ! ${PYTHON} "${PIEMAN_UTILS_DIR}"/check_redis.py -H "${REDIS_HOST}" -P "${REDIS_PORT}"; then
+            fatal "could not connect to Redis"
+
+            REDIS_IS_AVAILABLE=false
+
+            # Don't exit with non-zero code since the problem related to Redis
+            # mustn't affect the success of the current build.
+        else
+            info "Redis is available"
+        fi
+    fi
+}
+
+# Sends a request to Build Status Codes server.
+# Globals:
+#     PIEMAN_UTILS_DIR
+#     PYTHON
+# Arguments:
+#     None
+# Returns:
+#     None
+send_request_to_bsc_server() {
```

Comment:
Looks like you forget to list `REDIS_IS_AVAILABLE`, `ENABLE_BSC_CHANNEL` and `PROJECT_NAME` globs.